File: Languages/Java/GCD_Recursive.java
import java.util.Scanner;
import java.lang.*;

public class GCD_recursive {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        int a = scanner.nextInt();
        int b = scanner.nextInt();

        int result = GCD(a, b);
        System.out.println("Gcd of a and b is " + result);
    }

    private static int GCD(int a, int b) {
        if (a == 0) {
            return b;
        }
        if (b == 0) {
            return a;
        }
        if (a == b) {
            return a;
        }

        if (a > b) {
            return GCD(a - b, b);
        } else
            return GCD(a, b - a);
    }
}

File: Languages/Kotlin/GCD_Recursive.kt
import java.util.Scanner
import java.lang.*

object GCD_recursive {
    @JvmStatic
    fun main(args: Array<String>) {

        val scanner = Scanner(System.`in`)
        val a = scanner.nextInt()
        val b = scanner.nextInt()

        val result = GCD(a, b)
        println("Gcd of a and b is $result")
    }

    private fun GCD(a: Int, b: Int): Int {
        if (a == 0) {
            return b
        }
        if (b == 0) {
            return a
        }
        if (a == b) {
            return a
        }

        return if (a > b) {
            GCD(a - b, b)
        } else
            GCD(a, b - a)
    }
}